hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/GcsEventsHoodieIncrSource.java

```diff
@@ -145,6 +148,9 @@ public GcsEventsHoodieIncrSource(TypedProperties props, JavaSparkContext jsc, Sp
     this.gcsObjectDataFetcher = gcsObjectDataFetcher;
     this.queryRunner = queryRunner;
     this.schemaProvider = Option.ofNullable(schemaProvider);
+    this.snapshotLoadQuerySplitter = Option.ofNullable(props.getString(SNAPSHOT_LOAD_QUERY_SPLITTER_CLASS_NAME, null))
```

Comment:
Not setting `SNAPSHOT_LOAD_QUERY_SPLITTER_CLASS_NAME` could lead to NPE somewhere. Maybe just validate that the config is set and if not then Option.empty instead of Option.ofNullable. Then, wherever it is used handle empty if necessary.